File: data/src/main/java/com/todo/entity/Todo.java
package com.todo.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

@Entity
public class Todo {

	@Id
	@GeneratedValue
	private int id;

	@Column(nullable = false)
	private String text;

	private boolean completed;

	@JsonIgnore
	@ManyToOne
	private User user;

	public Todo() {
	}

	public Todo(String text, User user) {
		this.text = text;
		this.user = user;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getText() {
		return text;
	}

	public void setText(String text) {
		this.text = text;
	}

	public boolean isCompleted() {
		return completed;
	}

	public void setCompleted(boolean completed) {
		this.completed = completed;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}
}

File: frontend-ng/src/app/todo.ts
export class Todo {

  constructor(private _id: number, private _text: string, private _completed: boolean) {
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get text(): string {
    return this._text;
  }

  set text(value: string) {
    this._text = value;
  }

  get completed(): boolean {
    return this._completed;
  }

  set completed(value: boolean) {
    this._completed = value;
  }
}
